File: code/ch11/11.1.1_maxslidingwindow.cpp
class Solution {
public:
    vector<int> maxSlidingWindow(vector<int>& nums, int k) {
        vector<int> res = {};
        
        int length = nums.size();
        if (length * k == 0)
            return res;
        
        for (int i = 0; i < length - k + 1; i++) {
            int max_val = INT_MIN;
            
            for (int j = i; j < i + k; j++) {
                max_val = max(max_val, nums[j]);
            }
            
            res.push_back(max_val);
        }
        
        return res;
    }
};

File: code/ch11/11.1.1_maxslidingwindow.java
class Solution {
    public int[] maxSlidingWindow(int[] nums, int k) {
        if (k == 1)
            return nums;
        int turn = nums.length - k + 1;
        int[] res = new int[turn];
        for (int i = 0; i < turn; i++) {
            res[i] = getMax(nums, i, k);
        }
        return res;
    }

    private int getMax(int[] nums, int left, int k) {
        int max = nums[left];
        for (int i = 0; i < k-1; i++) {
            left++;
            max = Math.max(max, nums[left]);
        }
        return max;
    }
}